app/components/autocomplete/at_mention/at_mention.js

```diff
@@ -148,28 +147,66 @@ export default class AtMention extends Component {
         }
 
         if (nextProps.requestStatus !== RequestStatus.STARTED) {
-            const membersInChannel = nextProps.autocompleteUsersInCurrentChannel.in_channel || [];
+            let membersInChannel = nextProps.autocompleteUsersInCurrentChannel.in_channel || [];
             const membersOutOfChannel = nextProps.autocompleteUsersInCurrentChannel.out_of_channel || [];
-            const data = {
-                inChannel: membersInChannel,
-                notInChannel: membersOutOfChannel
-            };
+
+            // filter the current user out
+            membersInChannel = membersInChannel.filter((m) => m.id !== this.props.currentUserId);
+
+            let data = {};
+            if (membersInChannel.length > 0) {
+                data = Object.assign({}, data, {inChannel: membersInChannel});
+            }
+            if (membersOutOfChannel.length > 0) {
+                data = Object.assign({}, data, {notInChannel: membersOutOfChannel});
+            }
+            if (data.inChannel || data.notInChannel || this.checkSpecialMentions(matchTerm)) {
+                data = Object.assign({}, data, {specialMentions: this.getSpecialMentions()});
```

Comment:
The special mentions should be shown only if the `@` character is entered or the text matches one of them.

Also the sections should match the webapp and the order is:
* Channel Members
* Special Mentions
* Not in Channel